consensus/istanbul/core/core.go

```diff
@@ -92,6 +92,14 @@ type CoreBackend interface {
 
 	// ParentBlockValidators returns the validator set of the given proposal's parent block
 	ParentBlockValidators(proposal istanbul.Proposal) istanbul.ValidatorSet
+
+	// IsValidating return true if this instance is validating
+	IsValidating() bool
+
+	// IsElectedValidator returns true if instance is an elected validator
+	IsElectedValidator() bool
+
+	UpdateReplicaState() bool
```

Comment:
same

Reply:
`UpdateReplicaState` does get used in `startNewRound`. That goal of that is to make sure that the core checks that it's the primary for the sequence prior to starting that sequence.